backend/src/api/controllers/user-managment.controller.ts

```diff
@@ -21,31 +22,47 @@ import { UserMissingDataForm } from '~/common/forms/user.forms';
 import { convertForUserList } from '~/common/utils/user.util';
 import { IListUser, ShortUser } from '~/common/models/user/user';
 import { toShortUser } from '~/common/mappers/user.mapper';
+import { badRequestError } from '~/common/errors';
+import UserRepository from '~/data/repositories/user.repository';
+import CompanyRepository from '~/data/repositories/company.repository';
 
 type RegistrationUserProps = {
   host: string;
   origin: string;
   email: string;
   roleType: RoleType;
-  companyId: Company['id'];
+  userId: string;
 };
 
 const registerUserController = async ({
   host,
   origin,
   email,
   roleType,
-  companyId,
+  userId,
 }: RegistrationUserProps): Promise<IListUser> => {
-  const user = await registerUser(
+  const companyRepository = getCustomRepository(CompanyRepository);
+  const userRepository = getCustomRepository(UserRepository);
+
+  const user = await userRepository.geUserById(userId);
+
+  if (!user.company) {
+    throw badRequestError('User doesn`t create company!!!');
+  }
+
+  const company: Company = await companyRepository.findOne({
+    id: user.company.id,
+  });
+
+  const newUser = await registerUser(
     createDefaultUser(email),
     roleType,
-    companyId,
+    company.id,
```

Comment:
you can use here `user.company.id` as you already checked it is not null. no need to query again